[Automatic Migrations] UI - Fix data input steps

angorayc opened the pull request:
## Summary

https://github.com/elastic/kibana/pull/246028#pullrequestreview-3577554440




https://github.com/user-attachments/assets/36ca168b-fc12-44c6-9740-86feee5c9b83



<img width="854" height="1135" alt="Screenshot 2025-12-20 at 21 28 45" src="https://github.com/user-attachments/assets/6cc787af-8f5c-4207-854f-3a48232594ab" />



<img width="850" height="239" alt="image (4)" src="https://github.com/user-attachments/assets/d6f046ed-3b9c-4d0f-8eaa-becfc8b12702" />







<!--ONMERGE {"backportTargets":["9.3"]} ONMERGE-->

angorayc committed:
set step on vendor changed

angorayc committed:
Merge branch 'main' of github.com:elastic/kibana into step-issue

angorayc committed:
copy

angorayc committed:
Merge branch 'main' of github.com:elastic/kibana into step-issue

angorayc committed:
fix steps

angorayc committed:
check missing resource on flyout close

angorayc committed:
show migration source helper text only when feature flag is enabled

angorayc committed:
i18n

angorayc committed:
Merge branch 'main' into step-issue

angorayc committed:
Merge branch 'main' into step-issue

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376416)
* Commit: bbea8fbff448224829dcd6560a4c9a0802e51b0a

### Failed CI Steps
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/376416#019b4ae7-e3d2-4861-9aa8-2ff2937fac73)

### History
* :broken_heart: [Build #375987](https://buildkite.com/elastic/kibana-pull-request/builds/375987) failed fbd86c3006a83f35750952fec9e08f26fb8043b7
* :broken_heart: [Build #375874](https://buildkite.com/elastic/kibana-pull-request/builds/375874) failed 4706ab43a5acc7e06732d1ffa61c492b094205da

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376416","number":376416,"commit":"bbea8fbff448224829dcd6560a4c9a0802e51b0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375987","number":375987,"commit":"fbd86c3006a83f35750952fec9e08f26fb8043b7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375874","number":375874,"commit":"4706ab43a5acc7e06732d1ffa61c492b094205da"}],"number":376416}
buildkite-pr-comment-->

angorayc committed:
update copy

angorayc committed:
Merge branch 'step-issue' of github.com:angorayc/kibana into step-issue

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376462)
* Commit: bf6aabfbcc0bf9f4e52f29e20143888a459cd414

### Failed CI Steps
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/376462#019b4b5c-d460-4ac3-a646-8692c9636196)
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/376462#019b4bcc-ec48-4ddc-a09f-d49e80c5955b)
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/376462#019b4c3d-8baf-41ee-b39c-31a6963208dc)
* [Jest Tests #8](https://buildkite.com/elastic/kibana-pull-request/builds/376462#019b4caf-1bcf-4dd1-83ce-751a191e5947)
* [Scout: [ platform / streams_app ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/376462#019b4b63-52d7-42c5-9900-597e59c3a7aa)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a4e11e75be94d91808788b4fcb09d7d985b96ad7) | [after](https://github.com/elastic/kibana/commit/bf6aabfbcc0bf9f4e52f29e20143888a459cd414) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 10.7MB | 10.7MB | +1.6KB |



### History
* :broken_heart: [Build #376416](https://buildkite.com/elastic/kibana-pull-request/builds/376416) failed bbea8fbff448224829dcd6560a4c9a0802e51b0a
* :broken_heart: [Build #375987](https://buildkite.com/elastic/kibana-pull-request/builds/375987) failed fbd86c3006a83f35750952fec9e08f26fb8043b7
* :broken_heart: [Build #375874](https://buildkite.com/elastic/kibana-pull-request/builds/375874) failed 4706ab43a5acc7e06732d1ffa61c492b094205da

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376462","number":376462,"commit":"bf6aabfbcc0bf9f4e52f29e20143888a459cd414"},{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376416","number":376416,"commit":"bbea8fbff448224829dcd6560a4c9a0802e51b0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375987","number":375987,"commit":"fbd86c3006a83f35750952fec9e08f26fb8043b7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375874","number":375874,"commit":"4706ab43a5acc7e06732d1ffa61c492b094205da"}],"number":376462}
buildkite-pr-comment-->

angorayc committed:
Merge branch 'main' into step-issue

logeekal commented on x-pack/solutions/security/plugins/security_solution/public/siem_migrations/dashboards/components/migration_status_panels/migration_ready_panel.tsx:
```diff
@@ -22,33 +21,48 @@ import { useMigrationDataInputContext } from '../../../common/components/migrati
 
+  // Check for missing resources when the flyout is closed
+  useEffect(() => {
+    if (!isFlyoutOpen && flyoutMigrationId && migrationStats.id === flyoutMigrationId) {
```
👍

logeekal reviewed (commented):
Thanks @angorayc for this fix. Changes are looking good and i am still reviewing but you will need to change below text as well.

Users can start the migration without uploading resources and when migration is completed, we ask them to upload the resources again. Here is the code : 

https://github.com/elastic/kibana/blob/5e41a334ed2b41e5662d57773d916b4a6773b57c/x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/migration_status_panels/migration_result_panel.tsx#L185 


<img width="1080" height="646" alt="Screenshot 2025-12-24 at 11 16 58" src="https://github.com/user-attachments/assets/468c5006-094b-4522-9e00-078390ccbf97" />

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376643)
* Commit: cf4067eccfa8129ba57112d693f9d66b4c388c8a

### Failed CI Steps
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/376643#019b4f9a-69d9-47fd-a4d1-e30139149121)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/376643#019b500b-074d-4bec-9624-ab7477324f23)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/376643#019b507b-473b-4a85-a057-82367ab5f814)
* [Jest Tests #11](https://buildkite.com/elastic/kibana-pull-request/builds/376643#019b50eb-d7ab-40b2-81d5-447d541df188)
* [FTR Configs #97](https://buildkite.com/elastic/kibana-pull-request/builds/376643#019b4f9a-6a60-4377-8cd1-bd6fb77eb1b7)
* [Scout: [ security / security_solution ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/376643#019b4f9d-d14d-4816-8f7a-7a35c7a5e40b)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ef618d6b0829b5c4c955d56c1123434b89470b74) | [after](https://github.com/elastic/kibana/commit/cf4067eccfa8129ba57112d693f9d66b4c388c8a) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 10.7MB | 10.7MB | +1.6KB |



### History
* :broken_heart: [Build #376462](https://buildkite.com/elastic/kibana-pull-request/builds/376462) failed bf6aabfbcc0bf9f4e52f29e20143888a459cd414
* :broken_heart: [Build #376416](https://buildkite.com/elastic/kibana-pull-request/builds/376416) failed bbea8fbff448224829dcd6560a4c9a0802e51b0a
* :broken_heart: [Build #375987](https://buildkite.com/elastic/kibana-pull-request/builds/375987) failed fbd86c3006a83f35750952fec9e08f26fb8043b7
* :broken_heart: [Build #375874](https://buildkite.com/elastic/kibana-pull-request/builds/375874) failed 4706ab43a5acc7e06732d1ffa61c492b094205da

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376643","number":376643,"commit":"cf4067eccfa8129ba57112d693f9d66b4c388c8a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376462","number":376462,"commit":"bf6aabfbcc0bf9f4e52f29e20143888a459cd414"},{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376416","number":376416,"commit":"bbea8fbff448224829dcd6560a4c9a0802e51b0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375987","number":375987,"commit":"fbd86c3006a83f35750952fec9e08f26fb8043b7"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375874","number":375874,"commit":"4706ab43a5acc7e06732d1ffa61c492b094205da"}],"number":376643}
buildkite-pr-comment-->

angorayc committed:
review

angorayc committed:
review

angorayc commented on x-pack/solutions/security/plugins/security_solution/public/siem_migrations/rules/components/data_input_flyout/steps/enhancements/enhancements_data_input.tsx:
```diff
@@ -162,6 +162,7 @@ const EnhancementsDataInputContent = React.memo<EnhancementsDataInputContentProp
       <EuiFlexGroup direction="column" gutterSize="m">
         <EuiFlexItem>
           <EuiText size="s">{i18n.ENHANCEMENTS_INSTRUCTIONS}</EuiText>
+          <EuiText size="s">{i18n.ENHANCEMENTS_HELPER_TEXT}</EuiText>
```
<img width="853" height="1138" alt="Screenshot 2025-12-24 at 18 48 41" src="https://github.com/user-attachments/assets/29f370ad-8095-4632-9c71-254a0da7df4b" />

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376741)
* Commit: 9c95f5c69259a2e466de9ad48da39f17311b5b9f

### Failed CI Steps
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/376741#019b51d2-c7db-4e63-9021-b85951a04cb9)
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/376741#019b5243-b020-4f2b-8b57-ad024a2ad590)
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/376741#019b52b3-d49a-42f4-a098-b9600ceef499)
* [Jest Tests #1](https://buildkite.com/elastic/kibana-pull-request/builds/376741#019b5324-7a5b-4cb7-b688-c18cf0432a76)
* [FTR Configs #30](https://buildkite.com/elastic/kibana-pull-request/builds/376741#019b51d2-c7fe-47f4-a490-6faf01b575c9)
* [FTR Configs #123](https://buildkite.com/elastic/kibana-pull-request/builds/376741#019b51d2-c853-4e9b-9146-c28dd9980f0a)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376741#019b51d2-c7fe-47f4-a490-6faf01b575c9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376741/jobs/019b51d2-c7fe-47f4-a490-6faf01b575c9/artifacts/019b51f5-1398-4ba0-b5cf-3f94e12b76fe) FTR Configs #<span></span>30 / task_manager scheduling and running tasks should schedule tasks with API keys if request is provided

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ef618d6b0829b5c4c955d56c1123434b89470b74) | [after](https://github.com/elastic/kibana/commit/9c95f5c69259a2e466de9ad48da39f17311b5b9f) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 10.7MB | 10.7MB | +1.7KB |



### History
* :broken_heart: [Build #376643](https://buildkite.com/elastic/kibana-pull-request/builds/376643) failed cf4067eccfa8129ba57112d693f9d66b4c388c8a
* :broken_heart: [Build #376462](https://buildkite.com/elastic/kibana-pull-request/builds/376462) failed bf6aabfbcc0bf9f4e52f29e20143888a459cd414
* :broken_heart: [Build #376416](https://buildkite.com/elastic/kibana-pull-request/builds/376416) failed bbea8fbff448224829dcd6560a4c9a0802e51b0a
* :broken_heart: [Build #375987](https://buildkite.com/elastic/kibana-pull-request/builds/375987) failed fbd86c3006a83f35750952fec9e08f26fb8043b7
* :broken_heart: [Build #375874](https://buildkite.com/elastic/kibana-pull-request/builds/375874) failed 4706ab43a5acc7e06732d1ffa61c492b094205da

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376741","number":376741,"commit":"9c95f5c69259a2e466de9ad48da39f17311b5b9f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376643","number":376643,"commit":"cf4067eccfa8129ba57112d693f9d66b4c388c8a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376462","number":376462,"commit":"bf6aabfbcc0bf9f4e52f29e20143888a459cd414"},{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376416","number":376416,"commit":"bbea8fbff448224829dcd6560a4c9a0802e51b0a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/375987","number":375987,"commit":"fbd86c3006a83f35750952fec9e08f26fb8043b7"}],"number":376741}
buildkite-pr-comment-->

angorayc committed:
Merge branch 'main' into step-issue

angorayc committed:
Merge branch 'main' into step-issue

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376827)
* Commit: 8e5afbbc656a1f7cfccad72b29514112afe7bd13

### Failed CI Steps
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/376827#019b5bd3-e3e3-48f9-9659-611b58f4768a)
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/376827#019b5c43-eef1-4a77-bc35-6e871a6c5dff)
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/376827#019b5cb4-8203-44e2-b2de-d50762d10a3a)
* [Jest Tests #17](https://buildkite.com/elastic/kibana-pull-request/builds/376827#019b5d25-2983-400a-a491-1ef56a1a2975)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5d968c71dda6f3efccea88a8df5b6bb1c7370a8a) | [after](https://github.com/elastic/kibana/commit/8e5afbbc656a1f7cfccad72b29514112afe7bd13) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 10.7MB | 10.7MB | +1.7KB |



### History
* :broken_heart: [Build #376778](https://buildkite.com/elastic/kibana-pull-request/builds/376778) failed e28226128f9e4a2a19595cac31a58cba9b555623
* :broken_heart: [Build #376741](https://buildkite.com/elastic/kibana-pull-request/builds/376741) failed 9c95f5c69259a2e466de9ad48da39f17311b5b9f
* :broken_heart: [Build #376643](https://buildkite.com/elastic/kibana-pull-request/builds/376643) failed cf4067eccfa8129ba57112d693f9d66b4c388c8a
* :broken_heart: [Build #376462](https://buildkite.com/elastic/kibana-pull-request/builds/376462) failed bf6aabfbcc0bf9f4e52f29e20143888a459cd414
* :broken_heart: [Build #376416](https://buildkite.com/elastic/kibana-pull-request/builds/376416) failed bbea8fbff448224829dcd6560a4c9a0802e51b0a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376827","number":376827,"commit":"8e5afbbc656a1f7cfccad72b29514112afe7bd13"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376778","number":376778,"commit":"e28226128f9e4a2a19595cac31a58cba9b555623"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376741","number":376741,"commit":"9c95f5c69259a2e466de9ad48da39f17311b5b9f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376643","number":376643,"commit":"cf4067eccfa8129ba57112d693f9d66b4c388c8a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376462","number":376462,"commit":"bf6aabfbcc0bf9f4e52f29e20143888a459cd414"}],"number":376827}
buildkite-pr-comment-->

angorayc committed:
Merge branch 'main' into step-issue

angorayc committed:
Merge branch 'main' into step-issue

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/376976)
* Commit: f136f739c2880d8403112a81f098a3c85f9148f8

### Failed CI Steps
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/376976#019b6aaa-1ba0-4bc8-9557-eb35497cc063)
* [FTR Configs #6](https://buildkite.com/elastic/kibana-pull-request/builds/376976#019b6ad0-fb69-4238-870c-59ead2decabf)
* [Scout: [ security / security_solution ] plugin](https://buildkite.com/elastic/kibana-pull-request/builds/376976#019b6aae-b2b5-471e-8424-a60935b4bc6d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376976#019b6aaa-1ba0-4bc8-9557-eb35497cc063) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376976/jobs/019b6aaa-1ba0-4bc8-9557-eb35497cc063/artifacts/019b6ad0-8c41-47db-8436-954dd8f20b51) FTR Configs #<span></span>6 / task_manager scheduling and running tasks should schedule a task with rrule with fixed time and dtstart
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/376976#019b6ad0-fb69-4238-870c-59ead2decabf) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/376976/jobs/019b6ad0-fb69-4238-870c-59ead2decabf/artifacts/019b6adc-3095-4d6f-a7c4-7c7a539677b2) FTR Configs #<span></span>6 / task_manager scheduling and running tasks should schedule a task with rrule with fixed time and dtstart

### History
* :broken_heart: [Build #376875](https://buildkite.com/elastic/kibana-pull-request/builds/376875) failed c6b10fd1d0a192088a7e720e01c5564b95be5f60
* :broken_heart: [Build #376827](https://buildkite.com/elastic/kibana-pull-request/builds/376827) failed 8e5afbbc656a1f7cfccad72b29514112afe7bd13
* :broken_heart: [Build #376778](https://buildkite.com/elastic/kibana-pull-request/builds/376778) failed e28226128f9e4a2a19595cac31a58cba9b555623
* :broken_heart: [Build #376741](https://buildkite.com/elastic/kibana-pull-request/builds/376741) failed 9c95f5c69259a2e466de9ad48da39f17311b5b9f
* :broken_heart: [Build #376643](https://buildkite.com/elastic/kibana-pull-request/builds/376643) failed cf4067eccfa8129ba57112d693f9d66b4c388c8a

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376976","number":376976,"commit":"f136f739c2880d8403112a81f098a3c85f9148f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376875","number":376875,"commit":"c6b10fd1d0a192088a7e720e01c5564b95be5f60"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376827","number":376827,"commit":"8e5afbbc656a1f7cfccad72b29514112afe7bd13"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376778","number":376778,"commit":"e28226128f9e4a2a19595cac31a58cba9b555623"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376741","number":376741,"commit":"9c95f5c69259a2e466de9ad48da39f17311b5b9f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376643","number":376643,"commit":"cf4067eccfa8129ba57112d693f9d66b4c388c8a"}],"number":376976}
buildkite-pr-comment-->

angorayc committed:
Merge branch 'main' into step-issue

tylersmalley committed:
Fix infinite loop in MigrationReadyPanel test mock

The test mock for useGetMissingResources was creating a new jest.fn()
inside mockImplementation, causing an unstable function reference on
each render. This triggered the useEffect dependency to re-run
continuously, creating an infinite loop and test timeout.

Fixed by defining the mock function outside mockImplementation to
maintain a stable reference across renders.

kibanamachine committed:
Changes from node scripts/notice

tylersmalley committed:
Fix dashboard test infinite loop and move useMissingResources to common

- Fix unstable mock function reference in dashboard migration_ready_panel
  test that caused infinite render loops (same issue as rules test)
- Move useMissingResources hook from rules/components to common/hooks
  since it's shared between rules and dashboards
- Update mock path in macros_data_input.test.tsx to use new location
- Add leading space before missing resources text in rules panel

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/377585)
* Commit: 4fe8754a4b17df6e0e4dbb739dff7a9a67afb386

### Failed CI Steps
* [Quick Checks](https://buildkite.com/elastic/kibana-pull-request/builds/377585#019b7fbb-1c74-46ee-8e98-600e40703987)

### History
* :broken_heart: [Build #377331](https://buildkite.com/elastic/kibana-pull-request/builds/377331) failed 88e5c11a0f762c0caa0ba4f1412a2b5a6911ef2d
* :broken_heart: [Build #376976](https://buildkite.com/elastic/kibana-pull-request/builds/376976) failed f136f739c2880d8403112a81f098a3c85f9148f8
* :broken_heart: [Build #376875](https://buildkite.com/elastic/kibana-pull-request/builds/376875) failed c6b10fd1d0a192088a7e720e01c5564b95be5f60
* :broken_heart: [Build #376827](https://buildkite.com/elastic/kibana-pull-request/builds/376827) failed 8e5afbbc656a1f7cfccad72b29514112afe7bd13
* :broken_heart: [Build #376778](https://buildkite.com/elastic/kibana-pull-request/builds/376778) failed e28226128f9e4a2a19595cac31a58cba9b555623

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377585","number":377585,"commit":"4fe8754a4b17df6e0e4dbb739dff7a9a67afb386"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/377331","number":377331,"commit":"88e5c11a0f762c0caa0ba4f1412a2b5a6911ef2d"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376976","number":376976,"commit":"f136f739c2880d8403112a81f098a3c85f9148f8"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376875","number":376875,"commit":"c6b10fd1d0a192088a7e720e01c5564b95be5f60"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376827","number":376827,"commit":"8e5afbbc656a1f7cfccad72b29514112afe7bd13"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/376778","number":376778,"commit":"e28226128f9e4a2a19595cac31a58cba9b555623"}],"number":377585}
buildkite-pr-comment-->

angorayc committed:
Merge branch 'main' into step-issue

logeekal committed:
move loading spinned to upload button

logeekal reviewed (approved):
Thanks for the changes @angorayc . I did the desk test and all steps are working well.  I do have one suggestion though.

When we close the flyout, there is a flicker as if `Migration Ready Panel` is re-mounting as can be seen in below video.

https://github.com/user-attachments/assets/e8a85495-ea43-48f4-97bd-fd3fdfb9a0e7

I have raised a small PR : https://github.com/angorayc/kibana/pull/6 which changes this behaviour and moved the loading status inside the `Upload Resources` Button.  

Advantage for that is seamless UX without much Layoutshift.

Disvantage is the loading state is only visible if there are already missing resources and button is visible. IMO this is okay. Let me know what you think.

https://github.com/user-attachments/assets/5abd7d4b-4567-47ea-8e84-966829a5651e

angorayc committed:
Merge pull request #6 from logeekal/suggestion/missing_resources_loading

[Automatic Migration] Moved loading status to Upload Resources button

angorayc committed:
Merge branch 'main' into step-issue

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.3|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/248249">](https://github.com/elastic/kibana/pull/248249)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 9.6.6 -->
